### ISSUE TITLE
Torch FX SubgraphMatcher Any / Oneof Patterns

### ISSUE THREAD

**Issue by Jeremalloch** [pytorch/pytorch#109539]
### 🚀 The feature, motivation and pitch

I'm using the [SubgraphMatcher](https://github.com/pytorch/pytorch/blob/aed9bee0413dac190452fbfa9ab2a44b6e6843f5/torch/fx/passes/utils/matcher_utils.py#L51) to match torch.fx graphs.  I'd like to have some sort of set of nodes functionality when authoring patterns (ie match a convolution -> batch norm -> {relu, gelu, ... any activation}), without having to author one pattern per node type.

This was [previously implemented](https://github.com/pytorch/pytorch/pull/82853), but [then backed out](https://github.com/pytorch/pytorch/pull/83922) (for reasons I did not see).  Is this on the roadmap, or is there an alternative way to achieve this functionality that I'm not aware of?

**Comment by SherlockNoMad** [comment]
We can reland this. 
cc @andrewor14

**Comment by cgerum-eai** [comment]
I would be very interested in this as well. Any news?
